src/pages/index.js

```diff
@@ -0,0 +1,375 @@
+const postModalExitButton = document.querySelector("#post-modal-close-btn");
+const viewImageModalCloseButton = document.querySelector(
+  "#view-image-modal-close-btn"
+);
+const postSubmitBtn = postModal.querySelector(".modal__submit-btn");
+const editAvatarExitButton = document.querySelector("#edit-avatar-close-btn");
+const deleteImageExitButton = document.querySelector(
+  "#close-btn-modal-close-btn"
+);
+const deleteImageDeleteButton = document.querySelector("#confirm-delete-btn");
+const deleteImageCancelButton = document.querySelector("#cancel-delete-btn");
+const editProfileSubmitButton = document.querySelector(
+  "#edit-profile-submit-btn"
+);
+
+// Card Template and List Elements
+const cardTemplate = document.querySelector("#card-template");
+const cardList = document.querySelector(".cards__list");
+
+// Event Listeners
+if (settingsButton) {
+  settingsButton.addEventListener("click", () => {
+    editModalNameInput.value = profileName.textContent;
+    editModalDescriptionInput.value = profileDescription.textContent;
+    resetValidation(
+      editFormElement,
+      [editModalNameInput, editModalDescriptionInput],
+      settings
+    );
+    openModal(editModal);
+  });
+}
+if (editModalExitButton) {
+  editModalExitButton.addEventListener("click", () => {
+    closeModal(editModal);
+  });
+}
+if (editFormElement) {
+  editFormElement.addEventListener("submit", (evt) =>
+    editProfile(evt, settings)
+  );
+}
+if (postButton) {
+  postButton.addEventListener("click", () => {
+    openModal(postModal);
+  });
+}
+if (editAvatarFormElement) {
+  editAvatarFormElement.addEventListener("submit", (evt) => {
+    editAvatar(evt);
+  });
+}
+if (editAvatarButton) {
+  editAvatarButton.addEventListener("click", () => {
+    openModal(editAvatarModal);
+  });
+}
+if (editAvatarExitButton) {
+  editAvatarExitButton.addEventListener("click", () => {
+    closeModal(editAvatarModal);
+  });
+}
+if (postModalExitButton) {
+  postModalExitButton.addEventListener("click", () => {
+    closeModal(postModal);
+  });
+}
+if (postFormElement) {
+  postFormElement.addEventListener("submit", (evt) => postCard(evt, settings));
+}
+if (viewImageModalCloseButton) {
+  viewImageModalCloseButton.addEventListener("click", () => {
+    closeModal(viewImageModal);
+  });
+}
+if (deleteImageExitButton) {
+  deleteImageExitButton.addEventListener("click", () => {
+    closeModal(deleteImageModal);
+  });
+}
+
+// Functions
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    closeModal(document.querySelector(".modal_opened"));
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+
+function editProfile(evt) {
+  evt.preventDefault();
+
+  editProfileSubmitButton.textContent = "Saving...";
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions